wrapanapi/systems/msazure.py

```diff
@@ -1077,6 +1077,35 @@ def list_compute_images(self):
         return self.resource_client.resources.list(
             filter="resourceType eq 'Microsoft.Compute/images'")
 
+    def list_compute_images_by_resource_group(self, resource_group=None, free_images=None):
+        """
+        Args:
+            resource_group (str): Name of the resource group
+            free_images (bool): Whether to collect image which do not have any resource(VM) linked to it
+        """
+        resource_group = resource_group or self.resource_group
+        image_list = list(self.resource_client.resources.list(filter=f"resourceType eq 'Microsoft.Compute/images' and resourceGroup eq '{resource_group}'"))
+
+        if not free_images:
+            return image_list
+
+        vm_list = self.list_vms(resource_group=resource_group)
+        if len(vm_list) == 0:
+            return image_list
+
+        image_used_by_vm = []
+        for vm_name in vm_list:
+            image_used_by_vm.append(self.compute_client.virtual_machines.get(resource_group_name=resource_group, vm_name=vm_name).storage_profile.image_reference.id)
+
+        images_with_no_resources = []
+        for image in image_list:
+            if image.id not in image_used_by_vm:
```

Comment:
non-blocking, singular -> plural since it is a list of image**s** obtained from a list of VM**s**.
```suggestion
        images_used_by_vms = []
        for vm_name in vm_list:
            images_used_by_vms.append(self.compute_client.virtual_machines.get(resource_group_name=resource_group, vm_name=vm_name).storage_profile.image_reference.id)

        images_with_no_resources = []
        for image in image_list:
            if image.id not in images_used_by_vms:
```

Reply:
done